torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -198,6 +202,102 @@ bool is_numpy_scalar(PyObject* obj) {
           PyArray_IsScalar(obj, Floating));
 }
 
+at::Tensor tensor_from_cuda_array_interface(PyObject* obj) {
+  PyObject *cuda_dict = PyObject_GetAttrString(obj, "__cuda_array_interface__");
+  if (cuda_dict == nullptr) {
+    throw TypeError("attribute `__cuda_array_interface__` must exist");
+  }
+
+  // In the following code block, we extract the values of `__cuda_array_interface__`
+  std::vector<int64_t> sizes;
+  ScalarType dtype;
+  npy_intp data_ptr;
```

Comment:
I don't think there's any reason to use a `npy_intp` here, a `void*` will express this perfectly well, I think.

Reply:
(Or maybe a `intptr_t`, if you actually need a signed integral type. I haven't read the rest of the code yet, so I don't know exactly what you need.)